nltk/tag/perceptron.py

```diff
@@ -154,10 +169,14 @@ def tag(self, tokens):
             tag = self.tagdict.get(word)
             if not tag:
                 features = self._get_features(i, word, context, prev, prev2)
-                tag = self.model.predict(features)
+                tag = self.model.predict(features, conf)
             output.append((word, tag))
+            
             prev2 = prev
-            prev = tag
+            if conf == True:
+                prev = tag[0]
+            else:
+                prev = tag
```

Comment:
If we make predict always return confidence, then we can avoid the if-else and do:

```
prev2 = prev
return tag
```